cfme/tests/infrastructure/test_quota_tagging.py

```diff
@@ -28,13 +40,22 @@ def template_name(provider):
 
 
 @pytest.fixture()
-def prov_data():
+def prov_data(vm_name, provisioning):
     return {
-        "catalog": {'vm_name': ''},
+        "catalog": {'vm_name': vm_name},
```

Comment:
**Required**

The use of these fixtures, and the updates made to them, are confusing to me.

1. This fixture uses the `vm_name` fixture, which will provide a random name.
2. custom_prov_data _also_ uses `vm_name` fixture, overriding the random name already set in `prov_data`
3. Below in the test case methods, they're using `vm_name` fixture and `custom_prov_data` fixture, then further overriding the random name set first by `prov_data`, then again by `custom_prov_data` - so at the test case level, we've now generated three random names and changed the name at every step, apparently for no reason.

At the very least, the intended design and flow of these fixtures needs to be documented in-line.

If possible, please sort out how these fixtures are used in the existing test case, and your new test case, and simplify this down as much as possible so we're not setting values multiple times throughout test setup for no reason.